drv/transceivers-server/src/main.rs

```diff
@@ -320,92 +313,108 @@ impl ServerImpl {
 ////////////////////////////////////////////////////////////////////////////////
 
 impl idl::InOrderTransceiversImpl for ServerImpl {
-    fn get_modules_status(
+    fn get_module_status(
         &mut self,
         _msg: &userlib::RecvMessage,
-    ) -> Result<ModulesStatus, idol_runtime::RequestError<TransceiversError>>
+    ) -> Result<ModuleStatus, idol_runtime::RequestError<TransceiversError>>
     {
-        Ok(self
-            .transceivers
-            .get_modules_status()
-            .map_err(TransceiversError::from)?)
+        let (mod_status, result) = self.transceivers.get_module_status();
+
+        match result.error().count() {
+            0 => Ok(mod_status),
+            _ => Err(RequestError::from(TransceiversError::FpgaError)),
+        }
```

Comment:
I have a slight preference for the following:
```suggestion
        if result.error().is_empty() {
            Ok(mod_status)
        } else {
            Err(RequestError::from(TransceiversError::FpgaError)),
        }
```

since it's a little more obvious, though it's one extra LOC.

(same applies for all the functions below if you also prefer it)

Reply:
I think the `is_empty()` option is better as it is clearer by returning a `bool` and not relying on `number != 0 == True`.